tests/test_integration.py

```diff
@@ -22,11 +23,62 @@
 )
 from cyclic_boosting.quantile_matching import quantile_fit_gamma, quantile_fit_nbinom, quantile_fit_spline, J_QPD_S
 from cyclic_boosting.utils import smear_discrete_cdftruth
+from cyclic_boosting.interaction_selection import select_interaction_terms_anova
 from tests.utils import plot_CB, costs_mad, costs_mse
 
 np.random.seed(42)
 
 
+def test_interactions_selection(prepare_data, feature_properties, features, is_plot):
+    X, y = prepare_data
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
+
+    best_interaction_term_features = select_interaction_terms_anova(X_train, y_train, feature_properties, 3, 5)
+
+    expected = [
+        ("PG_ID_3", "L_ID"),
+        ("PG_ID_3", "PROMOTION_TYPE"),
+        ("L_ID", "PROMOTION_TYPE"),
+        ("PG_ID_3", "L_ID", "dayofweek"),
+        ("PG_ID_3", "L_ID", "PROMOTION_TYPE"),
+    ]
+    assert best_interaction_term_features == expected
+
+    features_ext = features.copy()
+    features_ext += best_interaction_term_features
+
+    explicit_smoothers = {
+        ("dayofyear",): SeasonalSmoother(order=3),
+        ("price_ratio",): IsotonicRegressor(increasing=False),
+    }
+
+    plobs = [
+        observers.PlottingObserver(iteration=1),
+        observers.PlottingObserver(iteration=-1),
+    ]
+
+    CB_est = pipeline_CBPoissonRegressor(
+        feature_properties=feature_properties,
+        feature_groups=features_ext,
+        observers=plobs,
+        maximal_iterations=50,
+        smoother_choice=common_smoothers.SmootherChoiceGroupBy(
+            use_regression_type=True, use_normalization=False, explicit_smoothers=explicit_smoothers
+        ),
+    )
+    CB_est.fit(X_train, y_train)
+
+    if is_plot:
+        plot_CB("analysis_CB_iterfirst", [CB_est[-1].observers[0]], CB_est[-2])
+        plot_CB("analysis_CB_iterlast", [CB_est[-1].observers[-1]], CB_est[-2])
+
+    yhat = CB_est.predict(X_test)
+
+    mad = np.nanmean(np.abs(y_test - yhat))
```

Comment:
Just so that I understand correctly, this test shows that, with the interaction terms, the Poisson regressor model provides a better result than its non-interacting counterpart in `test_poisson_regression`? If so, maybe renaming this test `test_poisson_regression_feature_interactions` (or something of the sort) would make it clearer.

Reply:
Exactly. Ok, renamed it.